components/brave_sync/brave_profile_sync_service_impl.cc

```diff
@@ -787,6 +802,131 @@ void BraveProfileSyncServiceImpl::LoadSyncEntityInfo(
   }
 }
 
+bool BraveProfileSyncServiceImpl::IsOtherBookmarksFolder(
+    const jslib::SyncRecord* record) const {
+  auto bookmark = record->GetBookmark();
+  if (!bookmark.isFolder)
+    return false;
+
+  std::string other_node_object_id;
+  if (model_->other_node()->GetMetaInfo("object_id", &other_node_object_id) &&
+      record->objectId == other_node_object_id)
+    return true;
+
+  if (bookmark.order == tools::kOtherNodeOrder &&
+      bookmark.site.title == tools::GetOtherNodeName() &&
+      bookmark.site.customTitle == tools::GetOtherNodeName()) {
+    return true;
+  }
+
+  return false;
+}
+
+void BraveProfileSyncServiceImpl::ProcessOtherBookmarksFolder(
+    const jslib::SyncRecord* record,
+    bool* pass_to_syncer) {
+  std::string other_node_object_id;
+    // Save object_id for late joined desktop to catch up with current id
+    // iteration
+  if (!model_->other_node()->GetMetaInfo("object_id", &other_node_object_id) &&
+      record->action == jslib::SyncRecord::Action::A_CREATE) {
+    tools::AsMutable(model_->other_node())->SetMetaInfo("object_id",
+                                                        record->objectId);
+  } else {
+    // If late joined desktop has bookmarks in other_node before joining sync
+    // chain
+    if (other_node_object_id != record->objectId &&
+        other_node_object_id ==
+          tools::GenerateObjectIdForOtherNode(std::string())) {
+      tools::AsMutable(model_->other_node())->SetMetaInfo("object_id",
```

Comment:
Not a question or notice, just clarifying.
If we are here, then current device has the first iteration of other_bookmarks_object_id, but the desktop which created `record` has 2nd or 3rd or ... iteration of other_bookmarks_object_id.